channel/commands.go

```diff
@@ -314,6 +316,342 @@ func (server *Server) gmCommand(conn mnet.Client, msg string) {
 		}
 
 		player.setMP(int16(amount))
+
+	case "setMaxHP":
+		player, err := server.players.getFromConn(conn)
+		if err != nil {
+			conn.Send(packetMessageRedText(err.Error()))
+			return
+		}
+		if len(command) < 2 {
+			conn.Send(packetMessageRedText("Usage: /setMaxHP <amount>"))
+			return
+		}
+		val, err := strconv.Atoi(command[1])
+		if err != nil {
+			conn.Send(packetMessageRedText("Amount must be a number"))
+			return
+		}
+		if val < 1 {
+			conn.Send(packetMessageRedText("Max HP must be at least 1"))
+			return
+		}
+		player.setMaxHP(int16(val))
+		conn.Send(packetMessageNotice(fmt.Sprintf("Set Max HP to %d", val)))
+
+	case "setMaxMP":
+		player, err := server.players.getFromConn(conn)
+		if err != nil {
+			conn.Send(packetMessageRedText(err.Error()))
+			return
+		}
+		if len(command) < 2 {
+			conn.Send(packetMessageRedText("Usage: /setMaxMP <amount>"))
+			return
+		}
+		val, err := strconv.Atoi(command[1])
+		if err != nil {
+			conn.Send(packetMessageRedText("Amount must be a number"))
+			return
+		}
+		if val < 0 {
+			conn.Send(packetMessageRedText("Max MP cannot be negative"))
+			return
+		}
+		player.setMaxMP(int16(val))
+		conn.Send(packetMessageNotice(fmt.Sprintf("Set Max MP to %d", val)))
+
+	case "str", "dex", "int", "luk":
+		var (
+			target *Player
+			val    int
+			err    error
+		)
+
+		switch len(command) {
+		case 2:
+			target, err = server.players.getFromConn(conn)
+			if err != nil {
+				conn.Send(packetMessageRedText(err.Error()))
+				return
+			}
+			val, err = strconv.Atoi(command[1])
+			if err != nil {
+				conn.Send(packetMessageRedText("Amount must be a number"))
+				return
+			}
+		case 3:
+			target, err = server.players.getFromName(command[1])
+			if err != nil {
+				conn.Send(packetMessageRedText(err.Error()))
+				return
+			}
+			val, err = strconv.Atoi(command[2])
+			if err != nil {
+				conn.Send(packetMessageRedText("Amount must be a number"))
+				return
+			}
+		default:
+			conn.Send(packetMessageRedText("Usage: /str <amount> | /str <player> <amount> (same for /dex /int /luk)"))
+			return
+		}
+
+		if val < 0 {
+			conn.Send(packetMessageRedText("Stat cannot be negative"))
+			return
+		}
+
+		switch command[0] {
+		case "str":
+			target.str = int16(val)
```

Comment:
Minor: Should these be refactored as player struct member variables? Do we forsee needing to update this value from multiple locations?